packages/css-parser-algorithms/src/consume/consume-component-block-function.ts

```diff
@@ -0,0 +1,621 @@
+			continue;
+		}
+
+		const result = consumeComponentValue(ctx, tokens.slice(i));
+		i += result.advance;
+		value.push(result.node);
+	}
+}
+
+export class SimpleBlockNode {
+	type: ComponentValueType = ComponentValueType.SimpleBlock;
+
+	startToken: CSSToken;
+	endToken: CSSToken;
+	value: Array<ComponentValue>;
+
+	constructor(startToken: CSSToken, endToken: CSSToken, value: Array<ComponentValue>) {
+		this.startToken = startToken;
+		this.endToken = endToken;
+		this.value = value;
+	}
+
+	tokens(): Array<CSSToken> {
+		return [
+			this.startToken,
+			...this.value.flatMap((x) => {
+				if (isToken(x)) {
+					return x;
+				}
+
+				return x.tokens();
+			}),
+			this.endToken,
+		];
+	}
+
+	toString(): string {
+		const valueString = this.value.map((x) => {
+			if (isToken(x)) {
+				return stringify(x);
+			}
+
+			return x.toString();
+		}).join('');
+
+		return stringify(this.startToken) + valueString + stringify(this.endToken);
+	}
+
+	indexOf(item: ComponentValue): number | string {
+		return this.value.indexOf(item);
+	}
+
+	at(index: number | string) {
+		if (typeof index === 'number') {
+			if (index < 0) {
+				index = this.value.length + index;
+			}
+			return this.value[index];
+		}
+	}
+
+	walk(cb: (entry: { node: ComponentValue, parent: ContainerNode }, index: number | string) => boolean | void) {
+		let aborted = false;
+
+		this.value.forEach((child, index) => {
+			if (aborted) {
+				return;
+			}
+
+			if (cb({ node: child, parent: this }, index) === false) {
+				aborted = true;
+				return;
+			}
+
+			if ('walk' in child) {
+				if (child.walk(cb) === false) {
+					aborted = true;
+					return;
+				}
+			}
+		});
+
+		if (aborted) {
+			return false;
+		}
+	}
+
+	toJSON() {
+		return {
+			type: this.type,
+			startToken: this.startToken,
+			tokens: this.tokens(),
+			value: this.value.map((x) => x.toJSON()),
+		};
+	}
+
+	isSimpleBlockNode(): this is SimpleBlockNode {
+		return SimpleBlockNode.isSimpleBlockNode(this);
+	}
+
+	static isSimpleBlockNode(x: unknown): x is SimpleBlockNode {
+		if (!x) {
+			return false;
+		}
+
+		if (!(x instanceof SimpleBlockNode)) {
+			return false;
+		}
+
+		return x.type === ComponentValueType.SimpleBlock;
+	}
+}
+
+/** https://www.w3.org/TR/css-syntax-3/#consume-simple-block */
+export function consumeSimpleBlock(ctx: Context, tokens: Array<CSSToken>): { advance: number, node: SimpleBlockNode | UnclosedSimpleBlockNode } {
+	const endingTokenType = mirrorVariantType(tokens[0][0]);
+	if (!endingTokenType) {
+		throw new Error('Failed to parse, a mirror variant must exist for all block open tokens.');
+	}
+
+	const value: Array<ComponentValue> = [];
+
+	let i = 1;
+
+	// eslint-disable-next-line no-constant-condition
+	while (true) {
+		const token = tokens[i];
+		if (!token || token[0] === TokenType.EOF) {
+			ctx.onParseError({
+				message: 'Unexpected EOF while consuming a simple block.',
+				start: tokens[0][2],
+				end: tokens[tokens.length - 1][3],
```

Comment:
We get here more of these numbers too which tie into how these are stored. Flagging in case you want to tackle with the `4` too.